applications/OptimizationApplication/custom_utilities/filtering/explicit_filter.cpp

```diff
@@ -126,25 +126,34 @@ void ComputeWeightForAllNeighbors(
     double& rSumOfWeights,
     std::vector<double>& rListOfWeights,
     const FilterFunction& rFilterFunction,
-    const double VertexMorphingRadius,
+    const double Radius,
     const EntityPoint<TEntityType>& rDesignPoint,
     const std::vector<typename EntityPoint<TEntityType>::Pointer>& rNeighbourNodes,
+    const std::vector<typename EntityPoint<TEntityType>::Pointer>& rDampingNeighbourNodes,
     const IndexType NumberOfNeighbours,
     Expression const * const pExpression)
 {
     for (IndexType i = 0; i < NumberOfNeighbours; ++i) {
         const double domain_size = GetDomainSize(*rNeighbourNodes[i], pExpression);
-        const double filter_weight = rFilterFunction.ComputeWeight(rDesignPoint.Coordinates(), rNeighbourNodes[i]->Coordinates(), VertexMorphingRadius);
-
-        rListOfWeights[i] = TWeightComputationType::Compute(filter_weight, domain_size);
+        const double filter_weight = rFilterFunction.ComputeWeight(rDesignPoint.Coordinates(), rNeighbourNodes[i]->Coordinates(), Radius);
+        double damping_multiplier = 1.0;
+        if(rDampingNeighbourNodes.size()>0){
```

Comment:
can you please wrap this block to separate functions to give possebility to easily change the way we compute the damping_multiplier, in case, we want other ways (like using cos function or simple zero fields).

Reply:
@Igarizza Done, pls check it out